firmware/robot2015/src-ctrl/drivers/mcp23017/mcp23017.cpp

```diff
@@ -0,0 +1,281 @@
+/*  MCP23017 library for Arduino

```

Comment:
@jjones646, take a look at PR #236 - it's been sitting there a while unmerged b/c it's missing a couple small things, but I've seen it work on a board.  If you've checked that this one works well, we can go ahead and delete the PR for the other one.

One thing to note though is that this file is GPL.  Technically we're not supposed to use GPL'd stuff in an Apache'd project, although I don't think it's really an issue for us here.  http://www.apache.org/licenses/GPL-compatibility.html.